### ISSUE TITLE
Failing to auth against Jfrog Artifactory for private repositories

### ISSUE THREAD

**Issue by stickycode** [rusenask/keel#76]
I'm trying to do use Artifactory as docker store with polling using the Kubernetes provider and it seems the user is not coming through.

Its picking up the secret OK and shows the correct username in the log, but somehow does not seem to be sending it on correctly. I've tried using the api key and user/pass that are validated on the Artifactory side.

Using 90515df8f4b5bc95042f1bbeee05221ae6f27b8f2ded55286063d2a8e4b901c1 
Kube 1.5.7

Created the secrets using 
```
kubectl create secret docker-registry myregistrykey --docker-server=https://index.docker.io/v1/ --docker-username=<user> --docker-password=<pass> --docker-email=<email>
```
But I'm sure that is working fine

```
time="2017-08-02T02:23:47Z" level=debug msg="performing scan" 
time="2017-08-02T02:23:47Z" level=info msg="secrets.defaultGetter: secret looked up successfully" image=company-name-docker.jfrog.io/app-report-services namespace=delivery provider=kubernetes registry=company-name-docker.jfrog.io 
time="2017-08-02T02:23:47Z" level=info msg="registry client: getting digest" registry="https://company-name-docker.jfrog.io" repository=app-report-services tag=10.29 
time="2017-08-02T02:23:47Z" level=error msg="trigger.poll.RepositoryWatcher.addJob: failed to get image digest" error="Get https://company-name-docker.jfrog.io/v2/: http: non-successful response (status=403 body=\"{\\n  \\\"errors\\\" : [ {\\n    \\\"status\\\" : 403,\\n    \\\"message\\\" : \\\"Forbidden\\\"\\n  } ]\\n}\")" image="company-name-docker.jfrog.io/app-report-services:10.29" 
time="2017-08-02T02:23:47Z" level=error msg="trigger.poll.RepositoryWatcher.Watch: failed to add image watch job" error="Get https://company-name-docker.jfrog.io/v2/: http: non-successful response (status=403 body=\"{\\n  \\\"errors\\\" : [ {\\n    \\\"status\\\" : 403,\\n    \\\"message\\\" : \\\"Forbidden\\\"\\n  } ]\\n}\")" image_name="company-name-docker.jfrog.io/app-report-services:10.29" registry_username=namebuild 
time="2017-08-02T02:23:47Z" level=error msg="trigger.poll.manager: failed to start watching repository" error="Get https://company-name-docker.jfrog.io/v2/: http: non-successful response (status=403 body=\"{\\n  \\\"errors\\\" : [ {\\n    \\\"status\\\" : 403,\\n    \\\"message\\\" : \\\"Forbidden\\\"\\n  } ]\\n}\")" image="company-name-docker.jfrog.io/app-report-services:10.29" schedule="@every 1m"
```

**Comment by rusenask** [comment]
Hi! Haven't ever used but I was expecting this to fail for artifactory. Some questions:
1. Is this the same Artifactory that's available here https://www.jfrog.com/open-source/ ? 
and is hosted version == OSS one? https://www.jfrog.com/artifactory/free-trial/#Cloud

2. Seems that they don't have webhooks (https://www.jfrog.com/jira/browse/RTFACT-14103) or do they? It would be simpler solution.

**Comment by stickycode** [comment]
Yes the same, was planning to use polling.

**Comment by stickycode** [comment]
I can set up login for you if that would help. 

AFAICT there are no triggers/webhooks on artifactory at this stage. There are plans but there have been for a while.

**Comment by rusenask** [comment]
That would be really helpful, you can send any details to karolis.rusenas@gmail.com.

I will have a look at it later today, I imagine that has something to do with `ping` that client does upon registry client creation, Artifactory might not have that endpoint or it might have a different one.

**Comment by rusenask** [comment]
should be fixed in [0.4.4](https://github.com/rusenask/keel/releases/tag/0.4.4)

**Comment by stickycode** [comment]
The auth works against artifactory and its polling for versions now. 

There is a stray error message 

```
2017/08/05 23:31:26 registry failed ping request, error: Get https://practiv-veda-docker.jfrog.io/v2/: http: non-successful response (status=403 body="{\n  \"errors\" : [ {\n    \"status\" : 403,\n    \"message\" : \"Forbidden\"\n  } ]\n}")
```
